src/plugins/data/public/query/query_string/language_service/query_result.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import './_recent_query.scss';
+
+import { BehaviorSubject } from 'rxjs';
+import { EuiButtonEmpty, EuiPopover, EuiText, EuiPopoverTitle } from '@elastic/eui';
+
+import React, { useState, useEffect } from 'react';
+
+export interface QueryStatus {
+  status: string;
+  message?: {
+    error?: string;
```

Comment:
imo.

```
body: {
   message: string;
   statusCode?: number;
}
```

will make the code easier to access the regardless if type error or not because the message can contain the error message. we can also make it not optional requiring them to include the message and the code will become cleaner without permitting it to be undefined.

`body` becaue `message.statusCode` is a little bit confusing.

on the fence about `body` being optional or not because i think if it pushes a new subject and it had a body in the previous state but not in the present state then i can see the footer items being shifted around.

Reply:
fixed the type